src/main/java/lotto/model/Lotto.java

```diff
@@ -38,6 +43,31 @@ public Set<Integer> numbers() {
     }
 
 
+    public Grade matchCount(Set<Integer> winningLotto, int bonusNumber) {
+        bonusContainValidate(winningLotto, bonusNumber);
+        return Grade.valueOf(this.numbers()
+                        .stream()
+                        .reduce(0,(count, lottoNumber) -> lottoMatchCount(winningLotto, count, lottoNumber))
+                , getBonusCheck(bonusNumber));
```

Comment:
stream의 `filter`를 활용하면 `lottoMatchCount()`를 사용하지 않고도 일치하는 번호 개수를 구할 수 있을 것 같아요.
```suggestion
        int count = (int) numbers().stream()
            .filter(number -> winningLotto.contains(number))
            .count();

        return Grade.valueOf(count, getBonusCheck(bonusNumber));
```